src/mainwindow.cpp

```diff
@@ -2766,6 +2766,16 @@ void MainWindow::updateNoteTextFromDisk(Note note) {
     ScriptingService::instance()->onCurrentNoteChanged(&currentNote);
 }
 
+void MainWindow::delayCheckNotesExists(const Note &note) {
+    int max_checks = 10;
+    while(!note.fileExists() && max_checks-->0) {
+        QTime sleepTime = QTime::currentTime().addSecs(1);
+        while(QTime::currentTime() < sleepTime) {
+            QCoreApplication::processEvents(QEventLoop::AllEvents, 100);
```

Comment:
This almost always is a really bad idea.

A better way might be to do it using QTimer's and recursion with a limit or do it in a separate thread.